src/utils/ethereum.ts

```diff
@@ -34,32 +48,37 @@ async function getTokens(wallet, hermezTokens) {
         type: "function",
       },
     ];
-    const balancePromises = hermezTokens.map((token) => {
+    const balancePromises: Promise<BigNumber>[] = hermezTokens.map((token) => {
       if (token.id === ETHER_TOKEN_ID) {
         // tokenID 0 is for Ether
         return provider.getBalance(ethereumAddress);
       } else {
         // For ERC 20 tokens, check the balance from the smart contract
         const contract = new ethers.Contract(token.ethereumAddress, partialERC20ABI, provider);
 
-        return (
-          contract
-            .balanceOf(ethereumAddress)
-            // We can ignore if a call to the contract of a specific token fails.
-            .catch(() => {})
-        );
+        // We can ignore if a call to the contract of a specific token fails.
+        // ToDo: Find a way to properly type the functions of the contract declared through the ABI.
+        // eslint-disable-next-line
+        const balance: Promise<BigNumber> = contract
+          .balanceOf(ethereumAddress)
```

Comment:
doesn't balanceOf already return a BigNumber?

Reply:
Don't blame me cause I may be completely lost here... Isn't `balanceOf` a feature of the smart contract? It's described in the `ABI` as a `uint256` so I assumed it was an unsigned integer. This is at least what I understand from the [docs](https://docs.soliditylang.org/en/develop/types.html#integers).

Reply:
uint256 is a type in Solidity, and then in JS it's a BigInt, but ethers wraps it in a BigNumber

Reply:
Yeah that makes sense. Line removed.